protocol/src/message/amqp/section.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{
+    message::amqp::codec::constants::MESSAGE_BODY_DATA,
+    message::amqp::{types::Binary, AmqpDecoder},
+    utils::TupleMapperSecond,
+};
+
+use super::{
+    codec::constants::{
+        MESSAGE_ANNOTATIONS, MESSAGE_APPLICATION_PROPERTIES, MESSAGE_BODY_SEQUENCE,
+        MESSAGE_BODY_VALUE, MESSAGE_DELIVERY_ANNOTATIONS, MESSAGE_FOOTER, MESSAGE_HEADER,
+        MESSAGE_PROPERTIES,
+    },
+    header::Header,
+    properties::Properties,
+    types::{
+        AmqpSequence, AmqpValue, ApplicationProperties, DeliveryAnnotations, Descriptor, Footer,
+        MessageAnnotations, Value,
+    },
+    AmqpDecodeError,
+};
+
+#[derive(Clone, Debug, PartialEq)]
+#[allow(clippy::large_enum_variant)]
+pub enum MessageSection {
+    Header(Header),
+    DeliveryAnnotations(DeliveryAnnotations),
```

Comment:
The AMQP 1.0 fields we support are:
```
 public Annotations Annotations { get; internal set; }

        public ApplicationProperties ApplicationProperties { get; set; }

        public Properties Properties { get; set; }

        public Data Data { get; }
````
other fields _must_ be read-only, so only for decoding, see the [.NET client](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/RabbitMQ.Stream.Client/Message.cs#L23-L29)

Other fields could cause incompatibility problems, see https://github.com/qweeze/rstream/issues/15

Reply:
In this case the section is for decoding. Exposed fields for write are in 

`MessageBuilder` to recap we should support only

- data
- message_annotations
- properties
- application properties

when creating a message. 

The rest : 

- header
- footer
- delivery annotations
- amqp value
- amq sequence

just for decoding  right?

Reply:
yes! that's correct!  
Thank you!

Reply:
>in this case the section is for decoding. 


A ok! yes I was looking for the encoding part!

Reply:
solved via https://github.com/rabbitmq/rabbitmq-stream-rust-client/pull/120/commits/4538d5f245c7781f9304675ad88e3ba286c095e5